beacon-chain/blockchain/receive_block.go

```diff
@@ -48,6 +49,9 @@ func (s *Service) ReceiveBlock(ctx context.Context, block interfaces.ReadOnlySig
 
 	s.cfg.ForkChoiceStore.Lock()
 	defer s.cfg.ForkChoiceStore.Unlock()
+	// Checks if the current blockRoot ( which is also the head root) is new.
+	// This check must come before forkchoiceUpdateWithExecution because it saves the new head.
+	isNewHead := s.isNewHead(blockRoot)
```

Comment:
Why do this here versus next to the usage?

Reply:
I need to get is New Head before forkchoiceUpdateWithExecution saves the head which happens inside the onblock. otherwise it's not new anymore.

Reply:
This doesn't work either: an incoming block will always have `isNewHead==true` here. You need to save the headroot here and then after the call to `onBlock` check the the new head.